File: Semestru 4/MPP/lab/java/lab3/Services/src/main/java/org/example/services/ServiceSignup.java
package org.example.services;

import org.example.domain.Child;
import org.example.domain.Event;
import org.example.domain.Signup;
import org.example.domain.Tuple;
import org.example.repository.interf.ISignupRepository;

import java.util.ArrayList;

public class ServiceSignup implements IServiceSignup {
    private ISignupRepository Repo;

    public ServiceSignup(ISignupRepository repo) {
        this.Repo = repo;
    }

    public Iterable<Signup> GetAll() {
        return this.Repo.FindAll();
    }

    public Signup AddSignup (Child child, Event event) {
        this.Repo.Save(new Signup(child, event));
        return this.Repo.FindById(new Tuple<>(child.GetId(), event.GetId()));
    }

    public Signup GetById (int childId, int eventId) {
        return this.Repo.FindById(new Tuple(childId, eventId));
    }

    public Iterable<Signup> GetByEvent (int eventId) {
        return this.Repo.GetAllByEventId(eventId);
    }

    public Iterable<Signup> GetByChild (int childId) {
        return this.Repo.GetAllByChildId(childId);
    }

    public Signup Delete (int childId, int eventId) {
        return this.Repo.Delete(new Tuple(childId, eventId));
    }

    public Iterable<Signup> GetAllMapped(IServiceChild serviceChild, IServiceEvent serviceEvent) {
        var result = new ArrayList<Signup>();
        for (var signupNull : this.Repo.FindAll()) {
            result.add(
                    (Signup) new Signup(
                            serviceChild.GetById(signupNull.GetId().GetFirst()),
                            serviceEvent.GetById(signupNull.GetId().GetSecond())
                    ).SetId(
                            signupNull.GetId()
                    )
            );
        }
        return result;
    }
}

File: Semestru 4/MPP/lab/.net/lab3/lab3/Server/service/ServiceSignup.cs
using Model.app.domain;
using Persistence.app.repo.@interface;
using Services.services;

namespace Server.app.service
{
	public class ServiceSignup : IServiceSignup
	{
		private ISignupRepository Repo;

		public ServiceSignup(ISignupRepository repo) =>
			this.Repo = repo;

		public IEnumerable<Signup> GetAll() =>
			this.Repo.GetAllJoin();

		public Signup? GetById(Tuple<int, int> id) =>
			this.Repo.GetById(id);

		public IEnumerable<Signup>? GetByChild(int id) =>
			this.Repo.GetAllByChildId(id);

		public IEnumerable<Signup>? GetByEvent(int id) =>
			this.Repo.GetAllByEventId(id);

		public Signup? Add (Child child, Event event_) =>
			this.Repo.Create(new Signup(child, event_));

		public Signup? Update(Signup signup) => null!;
	}
}
